Lib/glyphs2ufo/builder.py

```diff
@@ -672,6 +673,46 @@ def add_glyph_to_groups(kerning_groups, glyph_data):
         kerning_groups[group] = kerning_groups.get(group, []) + [glyph_name]
 
 
+def propagate_anchors(ufo):
+    """Copy anchors from parent glyphs' components to the parent."""
+
+    def get_anchor(glyph, name):
+        return next((a for a in glyph.anchors if a.name == name), None)
+
+    for parent in ufo:
+        added_here = {}
+
+        # don't propagate anchors for mark glyphs
+        if any(a.name.startswith('_') for a in parent.anchors):
+            continue
+
+        # try to get anchors from base (first) components
+        glyph = parent
+        transformation = Transform()
+        while glyph.components:
```

Comment:
When `glyph` is re-assigned, it's set to the existing `glyph`'s first component (and presumably we won't have an infinite chain of components, i.e. eventually we'll get to a glyph with no components).